File: app/models/category.py
from sqlalchemy import Column, DateTime, Integer, String, func
from app.core.db import Base


class Category(Base):
    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(String, nullable=True, index=True)
    name = Column(String, nullable=False, unique=True)
    type = Column(String, nullable=False)
    created_at = Column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

File: src/features/category/types/category.ts
import type { TransactionType } from '@/features/transaction/types/transaction'

export type Category = {
  id: number
  user_id: number
  name: string
  type: TransactionType
  created_at: string // ISO datetime
  updated_at: string // ISO datetime
}

export type CategoryBrief = {
  id: number
  name: string
}
